### PR TITLE
refactor pv e2e code to improve readability

### PR DESCRIPTION
**Pull request by jeffvance**
**What this PR does / why we need it**:
Moved the helper functions out of _persistent_volumes.go_ to a new file, _pvutil.go_, in order to improve readability and make it easier to add new tests.
Also, all pod delete code now calls the same helper function `deletePodWithWait`.

**Release note**:
```
NONE
```

### CODE REVIEW

**Comment by jeffvance** [comment]
@copejon @jsafrane PTAL. This should be quick...

**Comment by spxtr** [comment]
@k8s-bot bazel test this
#40526

**Comment by liggitt** [comment]
is it a straight move? hard to tel

**Comment by jeffvance** [comment]
@liggitt Yes straight move with one caveat: I consolidated use of a common func to delete-a-pod-and-wait-for-it-to-terminate. So, _kubelet.go_'s and _persistent_volumes-disruptive.go_ 's version of deleting a pod has been consolidated into _pvutil.go_. Since the namespace arg is not needed now, `ns` was removed from the calls to deletePod in _persistent_volumes.go_ .
moved the `deletePodwithWait` func from _kubelet.go_ to _pvutil.go_ and changed the code in _persistent_volumes.go_ and in _persistent_volumes-disruptive.go_ to call it rather than using their own, identical func to delete a pod.

File: test/e2e/persistent_volumes-disruptive.go [hunk 1]
```diff
@@ -222,7 +222,7 @@ func initTestCase(f *framework.Framework, c clientset.Interface, pvConfig persis
 
 // tearDownTestCase destroy resources created by initTestCase.
 func tearDownTestCase(c clientset.Interface, f *framework.Framework, ns string, pod *v1.Pod, pvc *v1.PersistentVolumeClaim, pv *v1.PersistentVolume) {
-	deletePod(f, c, ns, pod)
```
Comment (liggitt) [inline test/e2e/persistent_volumes-disruptive.go]: is namespace always set in pod and identical to ns?

File: test/e2e/pvutil.go [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+}
+
+// Deletes the passed-in pod and waits for the pod to be terminated. Resilient to the pod
+// not existing.
+func deletePodWithWait(f *framework.Framework, c clientset.Interface, pod *v1.Pod) {
+
+	if pod == nil {
+		return
+	}
+	framework.Logf("Deleting pod %v", pod.Name)
```
Comment (liggitt) [inline test/e2e/pvutil.go]: the old method no-oped on pods with no name

File: test/e2e/persistent_volumes-disruptive.go (hunk 1 above)
Comment (jeffvance) [inline test/e2e/persistent_volumes-disruptive.go]: Yes. The pod's ns is the same as used by the test and is set in the pod -- see `makePod()`

File: test/e2e/pvutil.go (hunk 2 above)
Comment (jeffvance) [inline test/e2e/pvutil.go]: Correct. However. the pods always have a generated name.

Comment (jeffvance) [inline test/e2e/pvutil.go]: @liggitt @saad-ali is there a label that can be applied to re-run the build/test? I looked at the 3 failures below and, unless I'm missing something, they appear to be build/test infra related. Thanks.

**Comment by liggitt** [comment]
the "Rerun command" in the bot comment is what you want

**Comment by liggitt** [comment]
@k8s-bot unit test this
@k8s-bot verify test this
@k8s-bot bazel test this

**Comment by jeffvance** [comment]
@k8s-bot cvm gce e2e test this

**Comment by jeffvance** [comment]
@liggitt all tests pass so can I get a /lgtm so we can get this merged? This pr will cause at least one pr in the queue to need to be rebased but that is expected. Thanks.

**Comment by jeffvance** [comment]
Rebased.

**Comment by jeffvance** [comment]
@liggitt Thanks for the lgtm. Can you become an assignee?  Thanks again!

**Comment by jsafrane** [comment]
@k8s-bot unit test this

**Pull request merged**